cpp/modmesh/view/RManager.cpp

```diff
@@ -281,6 +202,217 @@ void RManager::setUpMenu()
+
+    auto * move_camera_forward = new RAction(
+        QString("Move camera forward (Ctrl+W/⬆)"),
+        QString("Move camera forward (Ctrl+W/⬆)"),
+        createCameraTranslateItemHandler([](CameraInputState & input)
+                                         { input.tzAxisValue = 1.0; }));
+
+    auto * move_camera_backward = new RAction(
+        QString("Move camera down (Ctrl+S/⬇)"),
+        QString("Move camera down (Ctrl+S/⬇)"),
+        createCameraTranslateItemHandler([](CameraInputState & input)
+                                         { input.tzAxisValue = -1.0; }));
+
+    auto * rotate_camera_positive_yaw = new RAction(
+        QString("Rotate camera positive yaw"),
+        QString("Rotate camera positive yaw"),
+        createCameraRotateItemHandler([](CameraInputState & input)
+                                      { input.rxAxisValue = 1.0; }));
+
+    auto * rotate_camera_negative_yaw = new RAction(
+        QString("Rotate camera negative yaw"),
+        QString("Rotate camera negative yaw"),
+        createCameraRotateItemHandler([](CameraInputState & input)
+                                      { input.rxAxisValue = -1.0; }));
+
+    auto * rotate_camera_positive_pitch = new RAction(
+        QString("Rotate camera positive pitch"),
+        QString("Rotate camera positive pitch"),
+        createCameraRotateItemHandler([](CameraInputState & input)
+                                      { input.ryAxisValue = 1.0; }));
+
+    auto * rotate_camera_negative_pitch = new RAction(
+        QString("Rotate camera negative pitch"),
+        QString("Rotate camera negative pitch"),
+        createCameraRotateItemHandler([](CameraInputState & input)
+                                      { input.ryAxisValue = -1.0; }));
+
+    reset_camera->setShortcut(QKeySequence(Qt::Key_Escape));
+
+    m_viewMenu->addAction(reset_camera);
+    m_viewMenu->addAction(move_camera_up);
+    m_viewMenu->addAction(move_camera_down);
+    m_viewMenu->addAction(move_camera_right);
+    m_viewMenu->addAction(move_camera_left);
+    m_viewMenu->addAction(move_camera_forward);
+    m_viewMenu->addAction(move_camera_backward);
+    m_viewMenu->addAction(rotate_camera_positive_yaw);
+    m_viewMenu->addAction(rotate_camera_negative_yaw);
+    m_viewMenu->addAction(rotate_camera_positive_pitch);
+    m_viewMenu->addAction(rotate_camera_negative_pitch);
+}
+
+std::function<void()> RManager::createCameraTranslateItemHandler(const std::function<void(CameraInputState &)> & func) const
+{
+    return [this, func]()
+    {
+        try
+        {
+            auto * viewer = dynamic_cast<R3DWidget *>(m_mdiArea->currentSubWindow()->widget());
+
+            if (viewer->camera())
+            {
+                CameraInputState input{};
+                func(input);
+                viewer->cameraController()->updateCameraPosition(input, 0.01);
+            }
+        }
+        catch (std::bad_cast & e)
```

Comment:
You don't need to check `std::bad_cast` here; instead, you should check if it's `null_ptr`.

> If target-type is a pointer type, the result is the null pointer value of target-type.
If target-type is a reference type, an exception of a type that would match a [handler](https://en.cppreference.com/w/cpp/language/catch) of type [std::bad_cast](https://en.cppreference.com/w/cpp/types/bad_cast) is thrown.